src/cdk-experimental/column-resize/column-resize.ts

```diff
@@ -0,0 +1,109 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {AfterViewInit, ElementRef, NgZone, OnDestroy} from '@angular/core';
+import {Directionality} from '@angular/cdk/bidi';
+import {fromEvent, merge, ReplaySubject} from 'rxjs';
+import {filter, map, mapTo, pairwise, startWith, take, takeUntil} from 'rxjs/operators';
+
+import {_closest, _matches} from '@angular/cdk-experimental/popover-edit';
+
+import {ColumnResizeNotifier, ColumnResizeNotifierSource} from './column-resize-notifier';
+import {HEADER_CELL_SELECTOR, RESIZE_OVERLAY_SELECTOR} from './selectors';
+import {HeaderRowEventDispatcher} from './event-dispatcher';
+
+const HOVER_OR_ACTIVE_CLASS = 'cdk-column-resize-hover-or-active';
+const WITH_RESIZED_COLUMN_CLASS = 'cdk-column-resize-with-resized-column';
+
+let nextId = 0;
+
+/**
+ * Base class for ColumnResize directives which attach to mat-table elements to
+ * provide common events and services for column resizing.
+ */
+export abstract class ColumnResize implements AfterViewInit, OnDestroy {
```

Comment:
This should have a blank `@Directive()` decorator since it's using Angular features.

Reply:
Done